src/lib/peers/peer-connection/PeerConnection.ts

```diff
@@ -0,0 +1,239 @@
+import PeerInfo from '../PeerInfo';
+import { randomBytes } from 'crypto';
+import * as grpc from '@grpc/grpc-js';
+import PeerManager from '../PeerManager';
+import TurnClient from '../turn/TurnClient';
+import KeyManager from '../../keys/KeyManager';
+import { stringToProtobuf, protobufToString, firstPromiseFulfilled } from '../../utils';
+import { PeerClient } from '../../../../proto/compiled/Peer_grpc_pb';
+import { PeerMessage, SubServiceType } from '../../../../proto/compiled/Peer_pb';
+
+class PeerConnection {
+  private publicKey: string;
+  private keyManager: KeyManager;
+  private peerManager: PeerManager;
```

Comment:
Does it make sense for this class to be using `PeerManager` as a dependency? I know in VFS I did some circularity. But that's because I knew one was a weak reference.

Usually top level classes use bottom level classes as dependencies, not the other way around.

See `peers/PeerManager` and `peers/peer-connection/PeerConnection`.

Reply:
> Does it make sense for this class to be using PeerManager as a dependency?
Not at all, this is the cost of moving quickly. This will be cleaned up hopefully in the very next PR for validation modelling. We will be able to use our arkit dependency flow to assess cohesion and circularity.

Reply:
https://github.com/MatrixAI/js-polykey/issues/98